llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -121,18 +151,17 @@ void propagateAspectsThroughTypes(const TypesEdgesTy &Edges, const Type *Start,
 /// Time complexity: O((V + E) * T) where T is the number of input types
 /// containing aspects.
 void propagateAspectsToOtherTypesInModule(
-    const Module &M, TypeToAspectsMapTy &TypesWithAspects) {
+    const Module &M, TypeToAspectsMapTy &TypesWithAspects,
+    AspectValueToNameMapTy &AspectValues) {
   std::unordered_set<const Type *> TypesToProcess;
   const Type *DoubleTy = Type::getDoubleTy(M.getContext());
```

Comment:
Do we need to check if this could be NULL?

Reply:
I don't believe so. To my knowledge the context should be able to supply it in all cases we care about, but @AlexeySachkov may know better.

Reply:
> To my knowledge the context should be able to supply it in all cases we care about, but @AlexeySachkov may know better.

I don't see any guarantees in the doxygen documentation, but looking at the [source code](https://llvm.org/doxygen/Type_8cpp_source.html#l00227) we simply return an address of a [`LLVMContextImpl` field](https://llvm.org/doxygen/LLVMContextImpl_8h_source.html#l01464), which can't be `nullptr`.

We definitely don't need to insert a runtime `if` here and even assertion would excessive, I think.